src/ucs/datastruct/ptr_array.c

```diff
@@ -196,29 +198,41 @@ unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value,
 
     elem = &ptr_array->start[index];
 
-    /* Remove from free list and Populate */
+    /* Remove from free list and populate */
     ptr_array->freelist = ucs_ptr_array_freelist_get_next_params(*elem,
-        placeholder_p);
+                                                             &size_free_ahead);
     *elem = (uintptr_t)value;
 
-    ptr_array->n_occupied++;
-
     return index;
 }
 
 void ucs_ptr_array_remove(ucs_ptr_array_t *ptr_array, unsigned index,
                           uint32_t placeholder)
 {
     ucs_ptr_array_elem_t *elem = &ptr_array->start[index];
+    ucs_ptr_array_elem_t *next_elem;
+    uint32_t size_free_ahead;
 
     ucs_assert_always(!ucs_ptr_array_is_free(ptr_array, index));
 
-    ucs_ptr_array_freelist_element_set_free(elem, placeholder,
+    if (((index + 1) < (ptr_array)->size) &&
+         (__ucs_ptr_array_is_free((ptr_array)->start[index+1])) ) {
```

Comment:
still unaligned, remove trailing whitespace at 219 line
pls add whitespace before and after `+`

Reply:
Fixed.